thoth/storages/graph/postgres.py

```diff
@@ -4548,6 +4548,12 @@ def sync_adviser_result(self, document: dict) -> None:
             )
 
             # Output stacks - advised stacks
+            if not document["result"].get("report", {}):
```

Comment:
Also, the sync of unsolved packages will need to go before this if statement - there is no report created if there are unsolved packages found by resolver.

Reply:
But it depends on the same `document["result"].get("report", {})`
```
            # Mark down packages that were not solved if adviser run failed.
            for unresolved in document["result"].get("report", {}).get("_ERROR_DETAILS", {}).get("unresolved", []):
                self._create_python_package_version(
                    session,
                    package_name=unresolved,
                    package_version=None,
                    index_url=None,
                    os_name=None,
                    os_version=None,
                    python_version=None,
                    sync_only_entity=True,
                )
```

Reply:
But it depends on the same `document["result"].get("report", {})`

```
            # Mark down packages that were not solved if adviser run failed.
            for unresolved in document["result"].get("report", {}).get("_ERROR_DETAILS", {}).get("unresolved", []):
                self._create_python_package_version(
                    session,
                    package_name=unresolved,
                    package_version=None,
                    index_url=None,
                    os_name=None,
                    os_version=None,
                    python_version=None,
                    sync_only_entity=True,
                )
```